src/cunumeric/matrix/contract.cc

```diff
@@ -96,40 +110,51 @@ struct ContractImplBody<VariantKind::CPU, LegateTypeCode::DOUBLE_LT> {
 };
 
 template <>
-struct ContractImplBody<VariantKind::CPU, LegateTypeCode::HALF_LT> {
+struct ContractImplBody<VariantKind::CPU, LegateTypeCode::HALF_LT> : public ContractTypeHelper {
   void operator()(__half* lhs_data,
                   size_t lhs_ndim,
-                  int64_t* lhs_shape,
-                  int64_t* lhs_strides,
-                  int32_t* lhs_modes,
+                  len_type* lhs_shape,
+                  stride_type* lhs_strides,
+                  label_type* lhs_modes,
                   const __half* rhs1_data,
                   size_t rhs1_ndim,
-                  int64_t* rhs1_shape,
-                  int64_t* rhs1_strides,
-                  int32_t* rhs1_modes,
+                  len_type* rhs1_shape,
+                  stride_type* rhs1_strides,
+                  label_type* rhs1_modes,
                   const __half* rhs2_data,
                   size_t rhs2_ndim,
-                  int64_t* rhs2_shape,
-                  int64_t* rhs2_strides,
-                  int32_t* rhs2_modes)
+                  len_type* rhs2_shape,
+                  stride_type* rhs2_strides,
+                  label_type* rhs2_modes)
   {
     // TBLIS doesn't handle half-precision floating point directly, so we have to go through a
     // conversion to single-precision.
 
-    std::vector<int64_t> lhs_copy_strides(lhs_ndim);
-    int64_t lhs_size     = calculate_volume(lhs_ndim, lhs_shape, lhs_copy_strides.data());
+    // We're going to do some direct casts between TBLIS types and standard types. To be sure that
+    // these are safe, assert that the sizes are the same.
+    static_assert(sizeof(int64_t) == sizeof(len_type));
+    static_assert(sizeof(int64_t) == sizeof(stride_type));
+
+    std::vector<stride_type> lhs_copy_strides(lhs_ndim);
+    int64_t lhs_size =
+      calculate_volume(lhs_ndim, (int64_t*)lhs_shape, (int64_t*)lhs_copy_strides.data());
     float* lhs_copy_data = allocate_buffer(lhs_size);
-    half_tensor_to_float(lhs_copy_data, lhs_data, lhs_ndim, lhs_shape, lhs_strides);
+    half_tensor_to_float(
+      lhs_copy_data, lhs_data, lhs_ndim, (int64_t*)lhs_shape, (int64_t*)lhs_strides);
```

Comment:
same comment. we could use an overload of this function to deduplicate the changes for castings.

Reply:
done